### PR TITLE
Add entries to the glossary

### PR DESCRIPTION
**Pull request by Fabien-GELUS**
I propose to add some definitions to the glossary : smart contracts, chaincode, assets (and their attributes stored in the ledger) and nodes for example. Don't hesitate to share your own suggestions !

### CODE REVIEW

File: src/entry_points/glossary.md [hunk 1]
```diff
@@ -22,10 +22,14 @@
 
 - **AggregateTuple**: An AggregateTuple corresponds to the specification of an aggregation task of several model /model updates using an aggregate algo. It leads to the creation of one model / model update.
 
+- **Assets**: Assets are objects that define a ML project, and have to be implemented by an organization before being added in the Substra network. Substra registers, stores and organizes computations on four different kinds of Assets: Objectives, Datasets, Algorithms and, Models. These assets can be private or shared depending on their permission regime.
+
```
Comment (natct10) [inline src/entry_points/glossary.md:26]: Would you kindly add some links to the concepts in this document, please?

File: src/entry_points/glossary.md [hunk 2]
```diff
@@ -58,13 +62,19 @@
 
 - **Machine learning orchestration**: In contexts where multiple parties collaborate for elaborating machine learning models, the different operations (e.g. algorithms transfers, training computations, model evaluations, predictions…) need to be orchestrated in time and space. Such an orchestration is done over a network connecting the parties, and requires complete traceability of all operations, identities certifications, security (among others). Substra Framework enables the implementation of applications or services requiring secure, traceable, distributed machine learning orchestration. src: <https://www.substra.ai/en/faq>
 
+- **Metrics**:  The metrics corresponds to a `tar.gz/.zip` containing a `Dockerfile` and a Python script. The metric calculation script is used to quantify the accuracy of a model, during the evaluation of an [objective](#o). The organization has to implement the score method of this script which, given the predictions of a model and the true results, gives the accuracy score of the model.
+
```
Comment (natct10) [inline src/entry_points/glossary.md:66]: :+1:

File: src/entry_points/glossary.md [hunk 3]
```diff
@@ -58,13 +62,19 @@
...
 
 - **Model / Model Update**: A Model / Model Update is a potentially large file containing the parameters / update of parameters of a trained model. In the case of a neural network, a model would contain the weights of the connections. It is either the result of training an Algorithm with a given Dataset, corresponding to a training task (traintuple or composite_traintuple); or the result of an Aggregate Algorithm aggregating models or model updates;corresponding to an aggregation task (aggregate_tuple).
 
+## N
+
+- **Nodes**: Nodes are standalone computing and storage resources running the Substra code. They are organised into a network. It is assumed that independent partner organizations control their respective nodes. They form a private network, where every node is connected to all others.
```
Comment (natct10) [inline src/entry_points/glossary.md:73]: typo: organiZed

File: src/entry_points/glossary.md [hunk 4]
```diff
@@ -84,7 +94,8 @@
 
 ## S
 
-- **Smart contracts**: A smart contract defines the "transaction logic that controls the lifecycle of a business object contained in the world state. It is then packaged into a chaincode which is then deployed to a blockchain network." You can think of smart contracts as governing transactions, while chaincode is related to how smart contracts are packaged for deployment. [Source](https://hyperledger-fabric.readthedocs.io/en/release-2.0/smartcontract/smartcontract.html#smart-contract)
+- **Smart contracts**: Defines the rules between different organizations in executable code. Applications (like Substra backend) invoke a smart contract to generate transactions that are recorded on the ledger.  The permissions of each assets are implemented in trustless smart-contracts which filter the addition of traintuples in the ledger.
```
Comment (natct10) [inline src/entry_points/glossary.md:97]: type: double space here "ledger.  The"

File: src/entry_points/glossary.md (hunk 1 above)
Comment (Fabien-GELUS) [reply src/entry_points/glossary.md:26]: Is a link to the whitepaper sufficient ?

File: docs/_sources/entry_points/glossary.md.txt [hunk 5]
```diff
@@ -58,13 +62,19 @@
...
 
+## N
+
+- **Nodes**: Nodes are standalone computing and storage resources running the Substra code. They are organised into a network. It is assumed that independent partner organizations control their respective nodes. They form a private network, where every node is connected to all others.
+
 ## O
 
-- **Objective**: An objective correspond to a machine learning task on defined data types, a scientific question, which can be evaluated with a metrics on a test dataset. The metrics corresponds to a `tar.gz/.zip` containing a `Dockerfile` and Python scripts.
+- **Objective**: An objective correspond to a machine learning task on defined data types, a scientific question, which can be evaluated with a [metrics](#m) on a test dataset.
```
Comment (natct10) [inline docs/_sources/entry_points/glossary.md.txt]: Agree with that!

**Comment by Fabien-GELUS** [comment]
@natct10 Yes it seems nice to me !

**Pull request merged**